Avalonia.ListBoxAnimation.Samples/SelectingItemsControlExtension.cs

```diff
@@ -43,74 +42,153 @@ private static void SelectingItemsControlPropertyChanged(object? sender, Avaloni
-            isVerticalOffset
-                ? $"Vector3(this.FinalValue.X, this.FinalValue.Y{expression}, 0)"
-                : $"Vector3(this.FinalValue.X{expression}, this.FinalValue.Y, 0)"
-        );
-        offsetAnimation.InsertExpressionKeyFrame(1f, "this.FinalValue");
-        offsetAnimation.Duration = TimeSpan.FromMilliseconds(250);
-
-        // Create small scale animation so the pipe will "stretch" while it's moving
-        Vector3KeyFrameAnimation scaleAnimation = compositor.CreateVector3KeyFrameAnimation();
-        scaleAnimation.Target = "Scale";
-        scaleAnimation.InsertKeyFrame(0f, Vector3.One, quadraticEaseIn);
-        scaleAnimation.InsertKeyFrame(0.5f,
-            new Vector3(1f + (!isVerticalOffset ? 0.75f : 0f), 1f + (isVerticalOffset ? 0.75f : 0f), 1f),
-            quadraticEaseIn);
-        scaleAnimation.InsertKeyFrame(1f, Vector3.One, quadraticEaseIn);
-        scaleAnimation.Duration = TimeSpan.FromMilliseconds(250);
-
-        CompositionAnimationGroup compositionAnimationGroup = compositor.CreateAnimationGroup();
-        compositionAnimationGroup.Add(offsetAnimation);
-        compositionAnimationGroup.Add(scaleAnimation);
-        ImplicitAnimationCollection pipeVisualImplicitAnimations = compositor.CreateImplicitAnimationCollection();
-        float currentOffset = isVerticalOffset ? pipeVisual.Offset.Y : pipeVisual.Offset.X;
-        if (currentOffset == 0) // Visual first shown, offset not calculated, lets trigger using Offset
-            pipeVisualImplicitAnimations["Offset"] = compositionAnimationGroup;
-        else // Visual already shown, we can't trigger on Offset as it won't change
-            pipeVisualImplicitAnimations["Visible"] = compositionAnimationGroup;
-
-        pipeVisual.ImplicitAnimations = pipeVisualImplicitAnimations;
+        }
+
+        var isVertical = deltaPos is { X: 0 };
+        var dist = (float)Math.Abs(deltaPos.X + (float)deltaPos.Y);
+        var prevSize = prevInd.Bounds.Size;
+        var nextSize = nextInd.Bounds.Size;
+
+        var dir = (isVertical ? nextPos.Y : nextPos.X) > (isVertical ? prevPos.Y : prevPos.X);
+
+        ResetIndicator(prevInd);
+
+        PlayIndicatorAnimations(nextInd, isVertical, dir, dist, prevSize, nextSize);
+    }
+
+    private static void ResetIndicator(Visual? indicator, float opacity = 0)
+    {
+        if (indicator == null) return;
+        var visual = ElementComposition.GetElementVisual(indicator);
+        if (visual == null) return;
+        visual.Opacity = opacity;
+    }
+
+
+    private static void PlayIndicatorAnimations(Visual? indicator, bool isVertical, bool isForward, float distance,
+        Size beginSize, Size endSize)
+    {
+        if (indicator == null) return;
+        var visual = ElementComposition.GetElementVisual(indicator);
+        if (visual == null) return;
+
+        var comp = visual.Compositor;
+        var duration = TimeSpan.FromSeconds(0.6);
+        var size = indicator.Bounds.Size;
```

Comment:
To make sure this work on first time with virtualization (latest master) you need to fallback to the previous indicator size, e.g. `indicator.Bounds.Size.IsDefault ? beginSize : indicator.Bounds.Size`